### PR TITLE
Proposed additions for CPU reference materials

### PR DESCRIPTION
**Pull request by m00dawg**
Adds opcodes and basic (but not exhaustive) information about the 65C02 processor with the idea being all the 6502 and 65C02 opcodes and info is in one place (versus strewn all over the internet)

### CODE REVIEW

**Comment by mooinglemur** [comment]
Please resolve the file conflicts and mark as ready once completed

**Comment by m00dawg** [comment]
@mooinglemur I believe the conflicts should now be addressed. I ran into some cases where git didn't do what I expected so I manually brought changes over which should bring this in line with main (apart form the additions and chapter reorg)

File: X16 Reference - 03 - BASIC.md [hunk 1]
```diff
@@ -198,7 +198,7 @@ There are two ways to check the error channel from inside a program:
 40 CLOSE 15
 ```
 
-Refer to [Chapter 11](X16%20Reference%20-%2011%20-%20Working%20with%20CMDR-DOS.md#chapter-11-working-with-cmdr-dos) for more details on CMDR-DOS and the command channel.
+Refer to [Chapter 13](X16%20Reference%20-%2013%20-%20Working%20with%20CMDR-DOS.md) for more details on CMDR-DOS and the command channel.
```
Comment (mooinglemur) [inline X16 Reference - 03 - BASIC.md:201]: The anchor links are necessary for proper intra-PDF linkages

File: X16 Reference - 09 - VERA Programmer's Reference.md [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 
-Version 0.9
+This document describes the **V**ersatile **E**mbedded **R**etro **A**dapter or VERA. 
+Which was originally written and conceived by Frank van den Hoef. The core of this
+document has been forked from Version 0.9 of the VERA. The original documentation
+can be found [here](https://github.com/fvdhoef/vera-module/blob/rev4/doc/VERA%20Programmer's%20Reference.md).
 
-*Author: Frank van den Hoef*
+The Commander X16 uses a modified version of VERA which includes extra functionality,
+notabl the FX Aid additions. See [Chapter 10](X16%20Reference%20-%2010%20-%20VERA%20FX%20Reference.md)
```
Comment (mooinglemur) [inline X16 Reference - 09 - VERA Programmer's Reference.md:9]: notating typo (I'll commit a fix afterwards)

File: X16 Reference - 09 - VERA Programmer's Reference.md [hunk 3]
```diff
@@ -831,9 +834,11 @@ The FX feature set is available in VERA firmware version v0.3.1 or later. The Co
...
+FX features are controlled mainly by registers $9F29-$9F2C with DCSEL set to 2 through 6.  FX_CTRL ($9F29 w/ DCSEL=2) is the master switch for enabling or disabling FX behaviors.  When writing an application that uses FX, it is important that the FX mode be preserved and disabled in interrupt handlers in cases where the handler accesses VERA registers or VRAM, including the PSG sound registers. Reading from FX_CTRL returns the current state, and writing 0 to FX_CTRL suspends the FX behaviors so that the VERA can be accessed normally without mutating other FX state.
 
-Preliminary documentation for the feature can be found [here](VERA%20FX%20Reference.md#vera-fx-reference), but as this is a brand new feature, examples and documentation still need to be written.
+Preliminary documentation for the feature can be found in [Chapter 10]
+(X16%20Reference%20-%2010%20-%20VERA%20FX%20Reference.md), but as this is a brand new
+ feature, examples and documentation still need to be written.
```
Comment (mooinglemur) [inline X16 Reference - 09 - VERA Programmer's Reference.md:841]: line break broke link

File: X16 Reference - 10 - VERA FX Reference.md [hunk 4]
```diff
@@ -37,7 +37,7 @@ VERA is mapped as 32 8-bit registers in the memory space of the Commander X16, s
 		<td>$9F25</td>
 		<td>CTRL</td>
 		<td colspan="1" align="center">Reset</td>
-		<td colspan="6" align="center"><b>DCSEL</b><br><img src="images/redunderline.png"></td>
+		<td colspan="6" align="center"><b>DCSEL</b><br><img src="/images/redunderline.png"></td>
```
Comment (mooinglemur) [inline X16 Reference - 10 - VERA FX Reference.md:40]: are you sure we want absolute paths here? I think we should always stick to relative ones.

File: X16 Reference - 15 - Upgrade Guide.md [hunk 5]
```diff
@@ -47,7 +46,7 @@ Official community VERA bitstreams will be posted as releases at [X16Community/v
 |-|-|-|-|-|
 | Commander X16 | - | - | - | flashvera |
 | XGecu TL866II+ | Xgpro | minipro | minipro | - |
-| XGecu TL866-3G / T48 | [Xgpro](X16%20Reference%20-%20Appendix%20B%20-%20VERA%20recovery.md#using-a-windows-pc-to-upgrade-or-recover-a-vera) | - | - | - |
+| XGecu TL866-3G / T48 | [Xgpro](X16%20Reference%20-%20Appendix%20B%20-%20VERA%20Recovery.md) | - | - | - |
```
Comment (mooinglemur) [inline X16 Reference - 15 - Upgrade Guide.md:49]: anchor link missing for PDF

File: README.md [hunk 6]
```diff
@@ -1,4 +1,46 @@
...
+  * [Chapter 10: VERA FX Reference](X16%20Reference%20-%2010%20-%20VERA%20FX%20Reference.md)
+  * [Chapter 11: Sound Programming](X16%20Reference%20-%2011%20-%20Sound%20Programming.md)
+  * [Chapter 12: I/O Programming](X16%20Reference%20-%2012%20-%20IO%20Programming.md)
+  * [Chapter 13: Working with CMDR-DOS](X16%20Reference%20-%2013%20-%20Working%20with%20CMDR-DOS.md)
+  * [Chapter 14: Hardware Pinouts](X16%20Reference%20-%2014%20-%20Hardware.md)  
+  * [Chapter 15: Upgrade Guide](X16%20Reference%20-%2015%20-%20Upgrade%20Guide.md)
+  * [Appendix A: Sound](X16%20Reference%20-%20Appendix%20A%20-%20Sound.md)
+  * [Appendix B: VERA Recovery](X16%20Reference%20-%20Appendix%20B%20-%20VERA%20Recovery.md)
+  * [Appendix C: The 65C02 Processor](X16%20Reference%20-%20Appendix%20C%20-%2065C02%20Processor.md)
+  * [Appendix D: Official Expansion Cards](X16%20Reference%20-%20Appendix%20D%20-%20Official%20Expansion%20Cards.md)
```
Comment (mooinglemur) [inline README.md:37]: All of these links need anchor links for the PDF links to work properly

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -33,19 +33,24 @@ jobs:
...
+          "X16 Reference - Appendix D - Official Expansion Cards.md" \
           | md-to-pdf --config-file .gh/config.js --stylesheet .gh/markdown.css > "artifacts/Commander X16 Programmer's Reference Guide.pdf"
       - name: Create VERA Programmer's Reference PDF
         run: |
           export PATH=$(pwd)/node_modules/.bin:$PATH
           cat .gh/header_footer.md \
-          "VERA Programmer's Reference.md" \
-          "VERA FX Reference.md" \
+          "X16 Reference - 09 - VERA Programmer's Reference.md" \
+          "X16 Reference - 10 - VERA FX Reference.md" \
```
Comment (mooinglemur) [inline .github/workflows/build.yml:53]: I would remove the separate VERA build since it's part of the main PDF now

Comment (m00dawg) [reply .github/workflows/build.yml]: :+1:

File: X16 Reference - 10 - VERA FX Reference.md (hunk 4 above)
Comment (m00dawg) [reply X16 Reference - 10 - VERA FX Reference.md]: Fixed via markdown setting (though did convert some of those to markdown image tags)

**Review by mooinglemur — APPROVED** [review]
Good work!

Thanks for addressing all of the requests. I think this will help advance the docs a lot, especially when we are able to stand up a public version of the indexed/searchable documentation page you've been working on.

There are a bunch minor things like non-ideal page breaks in the PDF, such as a single row of a table spilling over.  I think we can address those later as we are starting to have a lot more changes behind this one.

**Pull request merged**